mono/mini/aot-runtime.c

```diff
@@ -5820,13 +5820,43 @@ mono_aot_get_unbox_trampoline (MonoMethod *method)
 
 	tinfo = mono_tramp_info_create (NULL, (guint8 *)code, 0, NULL, NULL);
 
+	// First trampoline's size is indicated by unbox_trampoline_p,
+	// and all trampolines are presumed to be the same size.
+	// FIXME Get the size of each trampoline from unwind info?
+
 	symbol_addr = read_unwind_info (amodule, tinfo, "unbox_trampoline_p");
 	if (!symbol_addr) {
 		mono_tramp_info_free (tinfo);
 		return FALSE;
 	}
 
-	tinfo->code_size = *(guint32*)symbol_addr;
+	guint32 const code_size = *(guint32*)symbol_addr;
+	tinfo->code_size = code_size;
+
+#ifdef HOST_AMD64
+
```

Comment:
Put this in tramp-amd64.c in a function named mono_arch_get_unbox_trampoline_size () or mono_amd64_get_unbox_trampoline_size (). Looks good otherwise.